functions/Find-DbaUserObject.ps1

```diff
@@ -37,292 +37,294 @@ You should have received a copy of the GNU General Public License
-                else
-                {
-                    $roles = $db.Roles | Where-Object {$_.IsSystemObject -eq 0 -and $_.Owner -eq $Pattern}
-                }
-                foreach ($role in $roles)
-                {
-                    $out = [PSCustomObject]@{
-                        ComputerName = $server.NetName
-                        SqlInstance = $server.InstanceName
-		                ObjectType = "Database Role" 
-                        ObjectOwner = $role.Owner
-                        ObjectName = $role.Name
-                        ObjectDetails = $db.Name
-                        }
-                        $out
-                }
-
-                ## assembly
-                if (-not $Pattern)
-                {             
-                    $Assemblies = $db.Assemblies | Where-Object {$_.IsSystemObject -eq 0 -and $_.Owner -ne "dbo"}
-                }
-                else
-                {
-                    $Assemblies = $db.Assemblies | Where-Object {$_.IsSystemObject -eq 0 -and $_.Owner -eq $Pattern}
-                }
-                foreach ($Assemblie in $Assemblies)
-                {
-                    $out = [PSCustomObject]@{
-                        ComputerName = $server.NetName
-                        SqlInstance = $server.InstanceName
-		                ObjectType = "Database Assembly" 
-                        ObjectOwner = $Assemblie.Owner
-                        ObjectName = $Assemblie.Name
-                        ObjectDetails = $db.Name
-                        }
-                        $out
-                }
-
-              ## synonyms
-                if (-not $Pattern)
-                {             
-                    $Synonymss = $db.Synonyms | Where-Object {$_.IsSystemObject -eq 0 -and $_.Owner -ne "dbo"}
-                }
-                else
-                {
-                    $Synonymss = $db.Synonyms | Where-Object {$_.IsSystemObject -eq 0 -and $_.Owner -eq $Pattern}
-                }
-                foreach ($Synonyms in $Synonymss)
-                {
-                    $out = [PSCustomObject]@{
-                        ComputerName = $server.NetName
-                        SqlInstance = $server.InstanceName
-		                ObjectType = "Database Synonyms" 
-                        ObjectOwner = $Synonyms.Owner
-                        ObjectName = $Synonyms.Name
-                        ObjectDetails = $db.Name
-                        }
-                        $out
-                }
-            }  
-
-        }
-    }
+	process
+	{
+		if ($Pattern -like "*\*")
+		{
+			Write-Verbose "Slash found in pattern"
+			$slashcount = ($Pattern -split "\\" | Measure-Object | Select-Object -ExpandProperty Count) - 1
+			if ($slashcount -eq 1)
+			{
+				Write-Verbose "Too few slashes, adding extra as required by regex"
+				$Pattern = $Pattern.Replace('\', '\\')
+				Write-Verbose "Pattern now $pattern"
+			}
+		}
+		
+		Write-Verbose "Starting"
```

Comment:
Inconsistent behavior and little extra value:
If you want to announce once that the function starts, move it to the begin Block.
If you want to mention it for each Instance it should process, move it inside the foreach loop and include the instance-name.